aten/src/ATen/core/ivalue.cpp

```diff
@@ -630,6 +631,17 @@ getClassConverter() {
   return classConverter;
 }
 
+void ivalue::Future::addCallbackWithTLSState(std::function<void(void)> callback) {
+    // Wrap cb into a function that runs with the current thread's TLS state.
+    at::ThreadLocalState tls_state;
+    std::function<void(void)> cbWithTlsState =
+        [tls_state = std::move(tls_state), callback = std::move(callback)] {
+          at::ThreadLocalStateGuard g(tls_state);
+          callback();
+        };
+    addCallback(cbWithTlsState);
```

Comment:
Would I be correct if I assume this will result in a copy (including `tls_state` and `callback`)?